app/controllers/supplemental_files_controller.rb

```diff
@@ -151,12 +151,15 @@ def edit_structure_path
     end
 
     def edit_file_information
-      ident = "machine_generated_#{params[:id]}".to_sym
-
-      if params[ident] && !@supplemental_file.machine_generated?
-        @supplemental_file.tags += ['machine_generated']
-      elsif !params[ident] && @supplemental_file.machine_generated?
-        @supplemental_file.tags -= ['machine_generated']
+      file_params = { machine_generated?: "machine_generated_#{params[:id]}".to_sym, caption_transcript?: "treat_as_transcript_#{params[:id]}".to_sym }
+
+      file_params.each do |k,v|
+        tag = k.to_s.gsub(/(caption_)?(.*)(\?)/, '\2')
```

Comment:
I think this would be easier to expand upon if instead of the regex `file_params`  turned into an array of hashes.  What do you think?
```suggestion
      file_params = [
        { param: "machine_generated_#{params[:id]}".to_sym, tag: "machine_generated", method: :machine_generated? },
        { param: "treat_as_transcript_#{params[:id]}".to_sym, tag: "transcript", method: :caption_transcript? }
      ]

      file_params.each do |v|
        param_name = v[:param]
        tag = v[:tag]
        method = v[:method]
```

Reply:
Yeah, that seems like it will be a better approach in the long term. Will rework it.